### PR TITLE
[CCFPCM-441] base auth guard

### PR DESCRIPTION
**Pull request by fw-noel**
[CCFPCM-441](https://bcdevex.atlassian.net/browse/CCFPCM-441)

Objective: 
Global auth guard applied to all requests but for healthcheck

### CODE REVIEW

File: apps/backend/src/common/guards/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import axios, { AxiosInstance } from 'axios';
+import { Request } from 'express';
+import jwt from 'jsonwebtoken';
+import { PUBLIC_ENDPOINT } from './../decorators/publicEndpoint.decorator';
+
+let axiosInstance: AxiosInstance;
```
Comment (chelsea-EYDS) [inline apps/backend/src/common/guards/auth.guard.ts:14]: Curious why we are defining axios instance outside of the class but assigning the value in the constructor?

**Review by chelsea-EYDS — APPROVED** [review]
Looks good! One question re: the axios instance in the auth guard... should this be an instance variable? Or if a  public field on the class should the value be assigned outside of the constructor? (I don't think it should affect the functionality but just curious )

**Comment by fw-noel** [comment]
Good catch!

**Review by chelsea-EYDS — APPROVED** [review]
Nice, looks great!

**Review by chelsea-EYDS — APPROVED** [review]

**Pull request merged**